test/p2p-test.js

```diff
@@ -8,6 +8,80 @@ const FullNode = require('../lib/node/fullnode');
 const {forValue} = require('./util/common');
 const packets = require('../lib/net/packets');
 
+describe('Pruned nodes', function() {
+  this.timeout(5000);
+
+  const node = new FullNode({
+    network: 'regtest',
+    memory: true,
+    listen: true,
+    port: 10000,
+    httpPort: 20000,
+    only: '127.0.0.1'
+  });
+
+  const node2 = new FullNode({
+    network: 'regtest',
+    memory: true,
+    listen: true,
+    prune: true
+  });
+
+  let peer;
+
+  before(async () => {
+    const waitForConnection = new Promise((resolve, reject) => {
+      node.pool.once('peer open', async (peer) => {
+        resolve(peer);
+      });
+    });
+
+    await node.open();
+    await node2.open();
+    await node.connect();
+    await node2.connect();
+    node.startSync();
+    node2.startSync();
+
+    peer = await waitForConnection;
+  });
+
+  after(async () => {
+    await node.close();
+    await node2.close();
+  });
+
+  const nodePackets = [];
```

Comment:
if you rebase your branch on master, you'll get the benefit of the improved p2p tests we just merged for BIP157. Take a look at how we collect incoming packets and check for packets of certain type.